nx.json

```diff
@@ -34,6 +51,33 @@
     "test:lib": {
       "outputs": ["{projectRoot}/coverage"],
       "inputs": ["default", "^public"]
+    },
+    "test:eslint": {
+      "inputs": ["default", "^public"]
+    },
+    "test:types": {
+      "outputs": [
+        "{projectRoot}/build/lib/**/*.d.ts",
+        "{projectRoot}/build/.tsbuildinfo"
+      ],
+      "inputs": ["default", "^public"],
+      "dependsOn": ["^test:types"]
+    },
+    "build:types": {
+      "outputs": ["{projectRoot}/build"],
+      "inputs": ["default", "^public"],
+      "dependsOn": ["^build:types"]
+    },
```

Comment:
Interestingly, `test:types` and `build:types` scripts tend to do the same things:

`test:types`: `tsc`
`build:types`: `tsc --build`

If I'm not mistaken, the build option is true by default, so these are equivalent. If we want to not write to disk on the `test:types` I think we can use: `test:types --build=false`, but we'll need to make sure that the builds we depend on happen first for this to work correctly!

Reply:
@TkDodo - if we want to make the adjustment above, can you create a new issue for this? And feel free to assign to me.

I'd like for this PR to be a like-for-like as much as possible so we don't change any behaviors!